game/src/main/java/net/demilich/metastone/game/spells/SummonSpell.java

```diff
@@ -286,7 +286,9 @@ protected void onCast(GameContext context, Player player, SpellDesc desc, Entity
 
 					if (context.getLogic().summon(player.getId(), minion, source, boardPosition, false)) {
 						summonedMinions.add(minion);
-						cards.remove(card);
+						if (!hasFilter || !desc.getBool(SpellArg.RANDOM_TARGET)) {
```

Comment:
just use `cards` for the soulbind effect and there, random target already works with replacement